src/dialogs/more-info/components/lights/dialog-light-color-favorite.ts

```diff
@@ -90,6 +93,31 @@ class DialogLightColorFavorite extends LitElement {
         : this._modes[0]);
   }
 
+  private _loadCurrentColor() {
+    const attributes = this.stateObj!.attributes;
+    const color_mode = attributes.color_mode;
+
+    if (attributes.color_mode === LightColorMode.XY) {
+      // XY color not supported for favorites. Try to grab the hs or rgb instead.
+      if (attributes.hs_color) {
+        this._color = { hs_color: attributes.hs_color };
+      } else if (attributes.rgb_color) {
+        this._color = { rgb_color: attributes.rgb_color };
+      }
+    } else if (
+      color_mode === LightColorMode.COLOR_TEMP &&
+      attributes.color_temp_kelvin
+    ) {
+      this._color = {
```

Comment:
We should also set the correct `LightPickerMode` `this._mode`

Reply:
Mmmm I guess `_updateModes` does that, I think that logic should be refactored a bit so it makes more sense (combined)

Reply:
Not sure I get what you're asking for. You want to fold `updateModes` and `loadCurrentColor` into a single function?

I don't know if I see much refactoring opportunity there, there doesn't seem like too much overlap between them.

Reply:
The part that sets the `_mode`, I found it confusing that the `_color` and `_mode` are set at different places, as they are related, and both determined by the `LightColorMode`.

Reply:
Ok, I tried an alternate implementation. I'm not sure if I feel this is much better or less confusing, but it does set _color and _mode more consistently.

Reply:
not really I think 😅  now we are calculating the current color always and only use it when `add == true`

Reply:
Well I do only want the current color when creating a new preset, because otherwise the current color of the bulb is not what we want when editing an existing preset. 

Let me know what's the path forward.

Reply:
Yeah, no I get that :-) I'll propose something tomorrow, it is really nitpicking anyway....